src/agent/Core/ApplicationPool/Group/ProcessListManagement.cpp

```diff
@@ -81,8 +81,13 @@ Group::findProcessWithStickySessionIdOrLowestBusyness(unsigned int id) const {
 
 	if (leastBusyProcessIndex == -1) {
```

Comment:
Good work noticing the contract description for route() which says we guarantee `process->canBeRoutedTo()`, I totally missed that.

Yes it is important to keep adhering to that contract, but I think we don't need to change the functions for the old routing algorithm that much. It's easier to change the code in route() in order to adapt to the differences in contract between the old routing functions and the new ones. Doing so should be less risky because we don't change the previous behavior so much.